pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java

```diff
@@ -213,7 +214,7 @@ public ZKMetadataStore(ZooKeeper zkc) {
                                     future.complete(new Stat(name, 0, 0, 0));
                                 } else if (code == Code.NODEEXISTS) {
                                     // We're emulating a request to create node, so the version is invalid
-                                    future.completeExceptionally(getException(Code.BADVERSION, path));
+                                    future.completeExceptionally(getException(Code.NODEEXISTS, path));
```

Comment:
This was done on purpose here. Since there is no "create()" operation but just the `put()` with the optional expected version. A `create()` is equivalent to a `put(Optional.of(-1))` (eg: expecting the node not to be there). In that light it makes sense to me to stick to the BadVersion. 

If the caller wants to make sure to create an entry, it just need to treat the bad version error for what it is: the state it's not what was expected.

Reply:
we can handle `BADVERSION` as `NODEEXISTS` at caller side. but my only concern was what exact benefit we can get  by returning different error-code. because actual error (in this case AlreadyExist) can help better in handling failure rather than deriving from `BADVERSION` . I will try to change handling at caller side but I think we might have usecases in future to explicitly know specific error to handle the failure.